### PR TITLE
prov/efa: adopt to rdma-core API adjustment

### PR DESCRIPTION
**Pull request by wzamazon**
There has been adjustment regarding rdma-core API
ibv_query_qp_data_in_order(). This patch adjust
the code accordingly.

### CODE REVIEW

**Comment by wzamazon** [comment]
@mrgolin

**Review by wenduwan — APPROVED** [review]

File: prov/efa/src/efa_base_ep.c [hunk 1]
```diff
@@ -354,8 +354,12 @@ int efa_base_ep_getname(fid_t fid, void *addr, size_t *addrlen)
...
 bool efa_base_ep_support_op_in_order_aligned_128_bytes(struct efa_base_ep *base_ep, enum ibv_wr_opcode op)
 {
-	return ibv_query_qp_data_in_order(base_ep->qp, op,
-					  IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES);
+	int caps;
+
+	caps = ibv_query_qp_data_in_order(base_ep->qp, op,
+					  IBV_QUERY_QP_DATA_IN_ORDER_RETURN_CAPS);
+
+	return caps == IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES;
```
Comment (shefty) [inline prov/efa/src/efa_base_ep.c:362]: I can pull this change into rc2.  Did libibverbs break the API here?

**Comment by shefty** [comment]
I have cherry-picked this fix into pr #8720 for rc2.

File: prov/efa/src/efa_base_ep.c (hunk 1 above)
Comment (mrgolin) [inline prov/efa/src/efa_base_ep.c:362]: I would do something like:
return !!(caps & IBV_QUERY_QP_DATA_IN_ORDER_ALIGNED_128_BYTES)

Comment (shefty) [reply prov/efa/src/efa_base_ep.c:362]: The proposed change makes sense.  Note that this PR and an equivalent patch has been merged into v1.18.x branch.  Any update would need to target both main and v1.18.x.

**Pull request merged**